crnk-rs/src/main/java/io/crnk/rs/JsonApiResponseFilter.java

```diff
@@ -92,4 +101,34 @@ private boolean isResourceResponse(Object response) {
 		boolean resourceList = ResourceListBase.class.isAssignableFrom(response.getClass());
 		return singleResource || resourceList;
 	}
+
+	/**
+	 * Uses {@link JsonApiRequestProcessor#isJsonApiRequest(HttpRequestContext)} to determine if the caller
+	 * requested a response in JSON-API format.
+	 *
+	 * @param requestContext the current container request context as received by this filter
+	 * @return <code>true</code>, if the caller requested a JSON-API response,<br />
+	 *     		<code>false</code>, otherwise
+	 */
+	private boolean isJsonApiRequest(ContainerRequestContext requestContext) {
+		HttpRequestContextBase httpRequestContextBase = new JaxrsRequestContext(requestContext, feature);
+		HttpRequestContextBaseAdapter httpRequestContext = new HttpRequestContextBaseAdapter(httpRequestContextBase);
+		return JsonApiRequestProcessor.isJsonApiRequest(httpRequestContext);
```

Comment:
You can specify the content type of a jaxrs method using `@Produces("application/vnd.api+json")`. This can then be queried via `ContainerResponseContext`'s `getMediaType()` method.

So, it is easiest to just check if the response context's media type is json-api and wrap the response if this is not the case yet. This way, no default has to be specified and the developer can simply specify if the response should be wrapped by adding a `@Produces` annotation.

Do you agree?